Test.Utility/TestEnvironmentProvider.cs

```diff
@@ -2,26 +2,29 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using Messerli.TempDirectory;
 
 namespace Messerli.Test.Utility
 {
     public class TestEnvironmentProvider : IDisposable
     {
         private const string ResourceDirectoryName = "Resources";
+        private const string DirectoryPrefix = "test-environment";
 
         private readonly IReadOnlyCollection<TestFile> _testFiles;
+        private readonly TempDirectory.TempDirectory _tempDirectory;
 
         public string RootDirectory { get; }
 
         public TestEnvironmentProvider(IReadOnlyCollection<TestFile> testFiles)
         {
             _testFiles = testFiles;
-
+            _tempDirectory = CreateTempDirectory();
+            RootDirectory = _tempDirectory.FullName;

```

Comment:
This could be a property with a getter fn.

Reply:
https://github.com/messerli-informatik-ag/test-utility/pull/7/commits/04aceaf2e3c7070c4eb11bb1e4dd2f498d3e5a36